addOns/client/src/main/java/org/zaproxy/addon/client/ClientHistoryTableModel.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.client;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.swing.table.AbstractTableModel;
+import org.parosproxy.paros.Constant;
+
+public class ClientHistoryTableModel extends AbstractTableModel {
+
+    private static final long serialVersionUID = 1L;
+
+    /** The column names. */
+    private static final String[] COLUMN_NAMES = {
+        Constant.messages.getString(ExtensionClientIntegration.PREFIX + ".history.table.header.id"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.timestamp"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.type"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.source"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.count"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.nodename"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.nodeid"),
+        Constant.messages.getString(
+                ExtensionClientIntegration.PREFIX + ".history.table.header.text")
+    };
+
+    /** The Constant defining the COLUMN COUNT. */
+    private static final int COLUMN_COUNT = COLUMN_NAMES.length;
+
+    private transient List<ReportedObject> history = new ArrayList<>();
```

Comment:
Sonarlint suggested transient?

I've been wondering about those suggestions 😄

Reply:
That might be misleading since we are not restoring/recreating the objects (but we don't serialize anyway).